plugin/src/main/java/org/opensearch/ml/rest/mcpserver/RestMLRegisterMcpToolsAction.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.rest.mcpserver;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.ml.plugin.MachineLearningPlugin.ML_BASE_URI;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.common.util.CollectionUtils;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.ml.common.spi.tools.Tool;
+import org.opensearch.ml.common.transport.mcpserver.action.MLMcpToolsRegisterOnNodesAction;
+import org.opensearch.ml.common.transport.mcpserver.requests.register.MLMcpToolsRegisterNodesRequest;
+import org.opensearch.ml.common.transport.mcpserver.requests.register.McpTools;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+import org.opensearch.transport.client.node.NodeClient;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+
+import lombok.extern.log4j.Log4j2;
+
+@ExperimentalApi
+@Log4j2
+public class RestMLRegisterMcpToolsAction extends BaseRestHandler {
+    private static final String ML_REGISTER_MCP_TOOLS_ACTION = "ml_register_mcp_tools_action";
+    private final Map<String, Tool.Factory> toolFactories;
+    private ClusterService clusterService;
+
+    /**
+     * Constructor
+     */
+    public RestMLRegisterMcpToolsAction(Map<String, Tool.Factory> toolFactories, ClusterService clusterService) {
+        this.toolFactories = toolFactories;
+        this.clusterService = clusterService;
+    }
+
+    @Override
+    public String getName() {
+        return ML_REGISTER_MCP_TOOLS_ACTION;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return ImmutableList.of(new Route(RestRequest.Method.POST, String.format(Locale.ROOT, "%s/mcp_tools/_register", ML_BASE_URI)));
+    }
+
+    @Override
+    public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        ActionRequestValidationException exception = new ActionRequestValidationException();
+        MLMcpToolsRegisterNodesRequest registerNodesRequest = getRequest(request);
+        if (CollectionUtils.isEmpty(registerNodesRequest.getMcpTools().getTools())) {
+            exception.addValidationError("tools list can not be null");
+            throw exception;
+        }
+        Set<String> buildInToolNames = toolFactories.values().stream().map(Tool.Factory::getDefaultType).collect(Collectors.toSet());
```

Comment:
Is it possible to register all of these built-in tools by default ? Why have to ask user register specific tools by themselves?

Reply:
Some tools needs init parameters like PPLTool needs model_id, we can't figure out this by default.

Reply:
Another option is let user pass in model_id for PPLTool. That way can avoid such register tool step, will make user experience much easier.